x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -99,7 +104,34 @@ public LogicalPlan optimize(LogicalPlan verified) {
         Batch label = new Batch("Set as Optimized", Limiter.ONCE,
                 new SetAsOptimized());
 
-        return Arrays.asList(substitutions, operators, constraints, operators, ordering, local, label);
+        return asList(substitutions, syntactic, operators, constraints, operators, ordering, local, label);
+    }
+
+    private static class AddMissingEquals extends OptimizerRule<Filter> {
+
+        @Override
+        protected LogicalPlan rule(Filter filter) {
+            // check the condition itself
+            Expression condition = replaceRawBoolFieldWithEquals(filter.condition());
+            // otherwise look for binary logic
+            if (condition == filter.condition()) {
+                condition = condition.transformUp(b ->
```

Comment:
Shouldn't this be enough? why do we need to check the top level separately?

Reply:
for cases such as `where bool` which have no other expression specified.

Reply:
I meant, that the transformUp would also catch this case, no?

Reply:
No because it filters on `BinaryLogic`.